contrib/rust/src/python/pants/contrib/rust/tasks/cargo_toolchain.py

```diff
@@ -0,0 +1,143 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import distutils.spawn
+import hashlib
+import os
+
+import requests
+from future.utils import PY3
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnit, WorkUnitLabel
+from pants.util.process_handler import subprocess
+
+from pants.contrib.rust.tasks.cargo_task import CargoTask
+
+
+class Toolchain(CargoTask):
+  LAST_KNOWN_FINGERPRINT = '40328ad8fa5cfc15cdb0446bb812a4bba4c22b5aee195cfb8d64b8ef1de5879c'
+
+  @classmethod
+  def supports_passthru_args(cls):
+    return True
+
+  @classmethod
+  def register_options(cls, register):
+    super(Toolchain, cls).register_options(register)
+    register(
+      '--toolchain',
+      type=str,
+      default=None,
```

Comment:
This should probably have a fixed default value to encourage build determinism. For now, it sounds like it would need to be a particular nightly sometime past `nightly-2018-12-31`. Once the `--build-plan` feature has landed, could swap it to a particular stable release.

It's just a default, and would want the docs to recommend overriding it. But important to have a stable default to minimize surprises.